### ISSUE TITLE
Clarification on floating point trailing zero behavior

### ISSUE THREAD

**Issue by summivox** [fmtlib/fmt#2675]
Source: https://github.com/fmtlib/fmt/blob/8.0.1/ChangeLog.rst

When I updated my project to 8.0.0 from 7.x, I noticed this:

-----------------

Formatting floating-point numbers no longer produces trailing zeros by default for consistency with std::format. For example:
```cpp
#include <fmt/core.h>

int main() {
  fmt::print("{0:.3}", 1.1);
}
```
prints "1.1". Use the '#' specifier to keep trailing zeros.

-------------------

Out of curiosity I tested it:

https://godbolt.org/z/KW1jfKaz1

```cpp
#include <fmt/core.h>

void Test(double x) {
    fmt::print(
        "{{0:.3}}   : {0:.3}\n"
        "{{0:.3f}}  : {0:.3f}\n"
        "{{0:#.3}}  : {0:#.3}\n"
        "{{0:#.3f}} : {0:#.3f}\n"
        "\n", x);
}

int main() {
    Test(1.1);
    Test(3);
    Test(0.25);
    Test(1.2345);
    return 0;
}
```

output for 7.1.3:
```
{0:.3}   : 1.10
{0:.3f}  : 1.100
{0:#.3}  : 1.10
{0:#.3f} : 1.100

{0:.3}   : 3.00
{0:.3f}  : 3.000
{0:#.3}  : 3.00
{0:#.3f} : 3.000

{0:.3}   : 0.250
{0:.3f}  : 0.250
{0:#.3}  : 0.250
{0:#.3f} : 0.250

{0:.3}   : 1.23
{0:.3f}  : 1.234
{0:#.3}  : 1.23
{0:#.3f} : 1.234
```

output for 8.0.1:
```
{0:.3}   : 1.1
{0:.3f}  : 1.100
{0:#.3}  : 1.10
{0:#.3f} : 1.100

{0:.3}   : 3
{0:.3f}  : 3.000
{0:#.3}  : 3.00
{0:#.3f} : 3.000

{0:.3}   : 0.25
{0:.3f}  : 0.250
{0:#.3}  : 0.250
{0:#.3f} : 0.250

{0:.3}   : 1.23
{0:.3f}  : 1.234
{0:#.3}  : 1.23
{0:#.3f} : 1.234
```

So basically only `g` is affected, while `f`, `#g`, `#f` are all unchanged?
How can we document this behavior (change) better?

**Comment by vitaut** [comment]
As the changelog entry says this affects the default format. So yes, the fixed format (`f`) etc. remain the same. If you have any documentation improvement suggestions a PR would be welcome.
